import java.sql.SQLException;
import java.util.Iterator;
import java.util.List;
import java.util.Map;
import java.util.Set;

import dao.DepartmentsDAO;


public class TestDAO {

	/**
	 * @param args
	 */
	public static void main(String[] args) throws SQLException {
		DepartmentsDAO dao = new DepartmentsDAO();

		/*
		// select
		int cnt = dao.selectDepartmentCountByName("Operations");
		System.out.println("Operations 부서 개수 : " + cnt);
		*/

		/*
		//insert
		HashMap map = new HashMap();
		map.put("id", "700");
		map.put("name", "개발부");
		dao.insertDepartment(map);
		*/
		
		/*
		// update
		DeptVO dvo = new DeptVO(700, "영업부");
		dao.updateDepartment(dvo);
		*/
		
		/*
		// delete
		dao.deleteDepartment(700);
		*/
		
		/*
		List list = dao.selectDeptByName("개발부");
		for (Object dvo : list) {
			System.out.println(dvo);
		}
		
		System.out.println("---------------------");
		Map map1 = dao.selectDeptByNameMap("개발부");
		Set keys = map1.keySet();
		Iterator itr = keys.iterator();
		while (itr.hasNext()) {
			Object key = itr.next();
			Object value = map1.get(key);
			System.out.println(key+"--"+value);		}
		*/

		// 이상하게 list2와 list3을 함께 구동시키면 오류가 발생한다!!
		// 아마 list2도 같은 형식으로 되야 하는게 아닌가 생각된다.
		// property와 column의 쌍이 맞아야 하는듯...
		// resultMap이 수정되어서 그렇다.
		/*
		System.out.println("---------------------");
		List list2 = dao.selectAllDept();
		for (Object obj : list2) {
			System.out.println(obj);
		}
		*/

		/*
		System.out.println("---------------------");
		List list3 = dao.selectDeptInfo();
		for (Object obj : list3) {
			System.out.println(obj);
		}
		*/
	}
}
